File: C#/BluffinMuffin.Protocol/Game/PlayerTurnEndedCommand.cs
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.DataTypes.Enums;

namespace BluffinMuffin.Protocol.Game
{
    /// <summary>
    /// Server informs all the clients that a player has taken action.
    /// </summary>
    public class PlayerTurnEndedCommand : AbstractGameCommand
    {
        /// <summary>
        /// The position on the table where is sitting the player that needs to play
        /// </summary>
        [ExampleValue(7)]
        public int PlayerPos { get; set; }
        /// <summary>
        /// The total amount of money a player have bet since the beginning of the round. 
        /// </summary>
        [ExampleValue(420)]
        public int PlayerBet { get; set; }
        /// <summary>
        /// The total amount of money a player have that has not been played.
        /// </summary>
        [ExampleValue(4200)]
        public int PlayerMoney { get; set; }
        /// <summary>
        /// The total amount of money currently played in this game. This include the money in the center, and the money played in front of each player.
        /// </summary>
        [ExampleValue(42000)]
        public int TotalPot { get; set; }
        /// <summary>
        /// The type of action that was taken by the player
        /// </summary>
        [ExampleValue(GameActionEnum.Call)]
        public GameActionEnum ActionType { get; set; }
        /// <summary>
        /// The amount of money that the player has put for this single action
        /// </summary>
        [ExampleValue(42)]
        public int ActionAmount { get; set; }
        /// <summary>
        /// The current state of the player that just played
        /// </summary>
        [ExampleValue(PlayerStateEnum.Playing)]
        public PlayerStateEnum State { get; set; }
    }
}

File: Python/BluffinMuffin.Protocol/game/player_turn_ended_command.py
from abstract_game_command import AbstractGameCommand
from data_types.enums.game_action_enum import GameActionEnum
from data_types.enums.player_state_enum import PlayerStateEnum

__author__ = 'ericmas001@gmail.com'



class PlayerTurnEndedCommand(AbstractGameCommand):
    def __init__(self, obj):
        super().__init__(obj)
        self.no_seat = obj['NoSeat']
        self.total_played_money_amount = obj['TotalPlayedMoneyAmount']
        self.total_safe_money_amount = obj['TotalSafeMoneyAmount']
        self.total_pot = obj['TotalPot']
        self.action_taken_type = GameActionEnum.parse(obj['ActionTakenType'])
        self.action_taken_amount = obj['ActionTakenAmount']
        self.player_state = PlayerStateEnum.parse(obj['PlayerState'])

    def __str__( self ):
        return '{0} ({1} {2}/{3} {4} {5}:{6} {7})'.format(
            super().__str__(),
            self.no_seat,
            self.total_played_money_amount,
            self.total_safe_money_amount,
            self.total_pot,
            GameActionEnum.to_string(self.action_taken_type),
            self.action_taken_amount,
            PlayerStateEnum.to_string(self.player_state)
        )
